pkg/agent/controller/traceflow/packetin.go

```diff
@@ -107,24 +107,39 @@ func (c *Controller) parsePacketIn(pktIn *ofctrl.PacketIn) (*opsv1alpha1.Tracefl
 	}
 
 	// Collect Service DNAT.
-	if pktIn.Data.Ethertype == 0x800 {
+	ctNwDst := ""
+	ipDst := ""
+	switch pktIn.Data.Ethertype {
+	case protocol.IPv4_MSG:
 		ipPacket, ok := pktIn.Data.Data.(*protocol.IPv4)
 		if !ok {
 			return nil, nil, errors.New("invalid traceflow IPv4 packet")
 		}
-		ctNwDst, err := getInfoInCtNwDstField(matchers)
+		ctNwDst, err = getInfoInCtNwDstField(matchers)
 		if err != nil {
 			return nil, nil, err
 		}
-		ipDst := ipPacket.NWDst.String()
-		if ctNwDst != "" && ipDst != ctNwDst {
-			ob := &opsv1alpha1.Observation{
-				Component:       opsv1alpha1.LB,
-				Action:          opsv1alpha1.Forwarded,
-				TranslatedDstIP: ipDst,
-			}
-			obs = append(obs, *ob)
+		ipDst = ipPacket.NWDst.String()
+	case protocol.IPv6_MSG:
+		ipPacket, ok := pktIn.Data.Data.(*protocol.IPv6)
+		if !ok {
+			return nil, nil, errors.New("invalid traceflow IPv6 packet")
+		}
+		ctNwDst, err = getInfoInCtIPv6DstField(matchers)
+		if err != nil {
+			return nil, nil, err
 		}
+		ipDst = ipPacket.NWDst.String()
+	default:
+		return nil, nil, errors.New("unsupported traceflow packet")
```

Comment:
How about "unsupported ether type for traceflow packet"?

Reply:
If there is something wrong in OVS, we might get unexpected packet here, and this branch is to handle when packet ethertype is not IP and IPv6.

Reply:
Understood. I meant could we include information about ether type check failure in the error message, so we know the exact reason for the error.

Reply:
Changed to:
`return nil, nil, fmt.Errorf("unsupported traceflow packet ether type %d", pktIn.Data.Ethertype)`